pycode/memilio-epidata/memilio/epidata/getNPIData.py

```diff
@@ -1006,117 +1053,80 @@ def get_npi_data(fine_resolution=2,
-                            indicator_code_active > 0)[0]
-                        if len(indicator_code_active_idx) > 0:
-                            # extract codes
-                            subcodes_nocombi = df_npis_combinations[code][1].loc[scidx, :]
-                            # only consider those codes which cannot be
-                            # combined; for these, values of 1 have to be
-                            # set to 0
-                            subcodes_nocombi = list(
-                                subcodes_nocombi
-                                [subcodes_nocombi == 0].index)
-                            # iterate over exclusive subcodes
-                            for subcode_excl in subcodes_nocombi:
-                                # iterate over less strict dynamic NPIs
-                                # i.e., where threshold is higher
-                                for level_other in level_lower:
-                                    # deactivate potential NPIs (with code:
-                                    # subcode_excl + level_other[1]) on days
-                                    # where NPI code_cols[scidx] + level[1]
-                                    # is active
-                                    if df_local_new.loc[indicator_code_active_idx, subcode_excl + level_other[1]].any():
-                                        print('Deactivating for ' + 'County ' + str(countyID) + ' and Incidence ' + str(level_other[0]))
-                                        print('\t' + npi_codes_prior_desc[npi_codes_prior[npi_codes_prior==subcode_excl + level_other[1]].index].values[0])
-                                        print('Due to Incidence > ' + str(level[0]) + ' and NPI ')
-                                        print('\t' + npi_codes_prior_desc[npi_codes_prior[npi_codes_prior==code_cols[scidx] + level[1]].index].values[0])
-                                        # print(list(df_local_new.loc[indicator_code_active_idx,'Date']))
-                                        print('\n')
-                                        df_local_new.loc[indicator_code_active_idx,
-                                                         subcode_excl + level_other[1]] = 0
-
-            # TODO: (maybe also remove incidence-dependent here? to discuss!)
-            # Remove conflicting non-incidence (?) dependent NPIs according to
-            # strictness index of Corona-Datenplattform and exclusion
-            # criteria defined in df_npis_combinations
-            for code in df_npis_combinations.keys():
-                # get other subcodes MX_Y_* below current main code MX_Y
-                subcodes_list = list(df_npis_combinations[code][0].keys())
+            # merge incidence dependent NPIs to have only one column for each subcode
+            df_merged = df_local_new.iloc[:, :2].copy()
+            for subcode in all_subcodes:
+                df_merged[subcode] = df_local_new.filter(
```

Comment:
probably filter is the better technique than str, but why didn't you actually use this filtering in `count_codes` in `df_local[df_local.NPI_code.str.contains(column)]`?

Reply:
we can use filter for both cases

Reply:
Sorry, not exactly true. We can use filter if the npi code is either the column or the index. Else it is easier to use str.contains. Also `df.filter(npi_code)` returns a dataframe and `df.columns.str.contains(regex=filter)` returns an array.